src/runtime_src/core/tools/xbmgmt2/SubCmdStatus.cpp

```diff
@@ -0,0 +1,113 @@
+  setIsDeprecated(_isDepricated);
+  setIsPreliminary(_isPreliminary);
+}
+
+void
+SubCmdStatus::execute(const SubCmdOptions& _options) const
+{
+  XBU::verbose("SubCommand: status");
+
+  XBU::verbose("Option(s):");
+  for (auto & aString : _options) {
+    std::string msg = "   ";
+    msg += aString;
+    XBU::verbose(msg);
+  }
+
+  // -- Retrieve and parse the subcommand options -----------------------------
+  std::string device = "all";
+  std::string report = "scan";
+  std::string format = "text";
+  std::string output;
+  bool help = false;
+
+  po::options_description queryDesc("Options");  // Note: Boost will add the colon.
+  queryDesc.add_options()
+    ("device,d", boost::program_options::value<decltype(device)>(&device), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest.  A value of 'all' (default) indicates that every found device should be examined.")
+    ("report,r", boost::program_options::value<decltype(report)>(&report), "The type of report to be produced. Reports currently available are:\n"
+                                                                           "  all         - All known reports are produced\n"
+                                                                           "  scan        - Terse report of found devices (default)\n"
+                                                                           "  electrical  - Voltages, currents, and power\n"
+                                                                           "                consumption on the device\n"
+                                                                           "  temperature - Temperatures across the device\n"
+                                                                           "  os-info     - Information relating to the operating\n"
+                                                                           "                system and drivers\n"
+                                                                           "  debug-ip    - Debug IP Status\n"
+                                                                           "  fans        - Fan status")
+    ("format,f", boost::program_options::value<decltype(format)>(&format), "Report output format. Valid values are:\n"
+                                                                           "  text        - Human readable report (default)\n"
+                                                                           "  json-2020.1 - JSON 2020.1 schema")
+    ("output,o", boost::program_options::value<decltype(output)>(&output), "Direct the output to the given file")
+    ("help,h", boost::program_options::bool_switch(&help), "Help to use this sub-command")
+  ;
+
+  // Parse sub-command ...
+  po::variables_map vm;
+
+  try {
+    po::store(po::command_line_parser(_options).options(queryDesc).run(), vm);
+    po::notify(vm); // Can throw
+  } catch (po::error& e) {
+    std::cerr << "ERROR: " << e.what() << std::endl << std::endl;
```

Comment:
two flushes?

Reply:
Understand your point here.

I am in the process of refactoring and cleaning up error conditions when exceptions are caught.  I will be sure to clean this up with future pull requests.